src/guides/v2.3/rest/tutorials/create-custom-rest-api.md

```diff
@@ -0,0 +1,572 @@
+
+*  `resource` sets the ACL resource that is required to have access to the endpoint. If you want to make it public (no authentication), you can use: `<resource ref="anonymous" />`
+
+## 4. Create Interfaces
+
+In this example, we have created interfaces for request and response
+
+`app/code/Dev/RestApi/Api/RequestItemInterface.php`:
+
+```php
+namespace Dev\RestApi\Api;
+
+interface RequestItemInterface
+{
+    const DATA_ID = 'id';
+    const DATA_DESCRIPTION = 'description';
+
+    /**
+     * @return int
+     */
+    public function getId();
+
+    /**
+     * @return string
+     */
+    public function getDescription();
+
+    /**
+     * @param int $id
+     * @return $this
+     */
+    public function setId(int $id);
+
+    /**
+     * @param string $description
+     * @return $this
+     */
+    public function setDescription(string $description);
+}
+```
+`app/code/Dev/RestApi/Api/ResponseItemInterface.php`:
+
+```php
+namespace Dev\RestApi\Api;
+
+interface ResponseItemInterface
+{
+    const DATA_ID = 'id';
+    const DATA_SKU = 'sku';
+    const DATA_NAME = 'name';
+    const DATA_DESCRIPTION = 'description';
+
+    /**
+     * @return int
+     */
+    public function getId();
+
+    /**
+     * @return string
+     */
+    public function getSku();
+
+    /**
+     * @return string
+     */
+    public function getName();
+
+    /**
+     * @return string|null
+     */
+    public function getDescription();
+
+    /**
+     * @param int $id
+     * @return $this
+     */
+    public function setId(int $id);
+
+    /**
+     * @param string $sku
+     * @return $this
+     */
+    public function setSku(string $sku);
+
+    /**
+     * @param string $name
+     * @return $this
+     */
+    public function setName(string $name);
+
+    /**
+     * @param string $description
+     * @return $this
+     */
+    public function setDescription(string $description);
+}
+```
+`app/code/Dev/RestApi/Api/ProductRepositoryInterface.php`:
+
+```php
+namespace Dev\RestApi\Api;
+
+interface ProductRepositoryInterface
+{
+    /**
+     * Return a filtered product.
+     *
+     * @param int $id
+     * @return \Dev\RestApi\Api\ResponseItemInterface
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getItem(int $id);
+
+    /**
+     * Set descriptions for the products.
+     *
+     * @param \Dev\RestApi\Api\RequestItemInterface[] $products
+     * @return void
+     */
+    public function setDescription(array $products);
+}
+```
+
+## 4. Create Models
+
+The objective of models to create classes that implement our interfaces and process data.
+In this example, we have created models for request, response and process data. In model we have defined two methods `getItem` which provide product details of the given product id and `setDescription` which updates the description of the given product item(s).
```

Comment:
```suggestion
In this example, we have created models for request, response and process data. In each model we have defined two methods `getItem`, which provides product details of the given product id, and `setDescription` which updates the description of the given product item(s).
```

Reply:
Hi @BarnyShergold, I have made changes, Kindly review it